src/First/README.md

```diff
@@ -0,0 +1,451 @@
+# First
+
+```haskell
+First a = First (Maybe a)
+```
+
+`First` is a `Monoid` that will always return the first, non-empty value when
+(2) `First` instances are combined. `First` is able to be a `Monoid` because
+it implements a `Maybe` under the hood. The use of the `Maybe` allows for an
+empty to be represented with a `Nothing`.
```

Comment:
What is "an empty"? Should this be "an empty _________ " (thing) or is "empty" supposed to be the thing we're talking about?

Reply:
an empty `First`. Say nothing meets the "criteria" like say an empty array when using `mconcat`, or if we are using `Maybe` to check a condition an neither of the values meet the criteria then there is no `First` to return, resulting in an empty `First` (`First( Nothing )`).

Basically for something to be a `Monoid` is must have a value the represents idenity for the operation in that anything `concat`-ed to the identity (or empty) will result in the same value. So `First(3).concat(First.empty())` results in `First( Just 3 )` as does `First.empty().concat(First(3))`

Any suggestions on how I can communicate that in this paragraph without too much theory?

Reply:
I think doing two things could really clear this up for beginners (like me!) and one thing applied everywhere at a later date could make the docs more accessible to beginners:

1. `empty` in the sentence sounds like a descriptor, so maybe using `empty 'First'` would help to clarify it
1. it doesn't have to include theory -- you can show values in a small example. I'm not sure if I'm doing these right, but here we go:
  ```js
  First.empty()                  // => First( Nothing )
  First(3)                       // => First( Just 3 )
  First(3).concat(First.empty()) // => First( Just 3 )
  First.empty().concat(First(3)) // => First( Just 3 )
  ```

* separately, all the category theory and ADTs are awesome for people that are familiar, but it'd also be potentially useful to point to the fantasy land spec or wikipedia or articles or something the first time a term is used in any doc.

Reply:
I like everything about this and think I can address these like this:
1. Add just like you suggest with empty `First` instance in the cases I speak about empty instance.
1. As the examples as described are shown in the `empty` static function, so what do you think about me adding a link to `#empty` where I have the word empty?
1. So I think that the docs could benefit from a section on basic category theory, typeclasses and signatures may address the reference issue.

Do you think that will do it?

Reply:
A link to the `empty` function would be real nice.

With regards to `3.`: if there are solid resources out there that can be linked to or copied+pasted+cited to handle this, then I would advise not recreating the wheel. However, if this could increase the accessibility and understanding of the project more than external links could and can provide more of a community benefit, then that's a noble thing _if you have the time_. Heck, maybe the docs get turned into a book.

Reply:
Makes sense